### PR TITLE
Adds a converter app for LDBC to CAPS

### PR DESCRIPTION
**Pull request by s1ck**

### CODE REVIEW

File: spark-cypher/src/main/scala/org/opencypher/spark/tools/LdbcConverter.scala [hunk 1]
```diff
@@ -0,0 +1,577 @@
...
+    RelType(Label("University", "Organisation"), "IS_LOCATED_IN", Label("City", "Place")) -> StructType(Seq(StructField("University_id", StringType, false), StructField("City_id", StringType, false))),
+    RelType(Label("Company", "Organisation"), "IS_LOCATED_IN", Label("Country", "Place")) -> StructType(Seq(StructField("Company_id", StringType, false), StructField("Country_id", StringType, false))),
+    RelType("Person", "IS_LOCATED_IN", "Place") -> StructType(Seq(StructField("Person_id", StringType, false), StructField("Place_id", StringType, false))),
+    RelType(Label("Person"), "IS_LOCATED_IN", Label("City", "Place")) -> StructType(Seq(StructField("Person_id", StringType, false), StructField("City_id", StringType, false))),
+    RelType("TagClass", "IS_SUBCLASS_OF", "TagClass") -> StructType(Seq(StructField("TagClass_id0", StringType, false), StructField("TagClass_id1", StringType, false)))
+  )
+
+  val relDictionary: Map[RelType, Int] = relMapping.zipWithIndex.map { case ((relType, _), i) => relType -> i }
```
Comment (Mats-SX) [inline spark-cypher/src/main/scala/org/opencypher/spark/tools/LdbcConverter.scala]: Could probably formulate this as `relMapping.keys.zipWithIndex.toMap`

File: spark-cypher/src/main/scala/org/opencypher/spark/tools/LdbcConverter.scala [hunk 2]
```diff
@@ -0,0 +1,577 @@
...
+          val withListProperty = tmpNodeDf
+            .join(propDF, tmpNodeDf.col(idColumnName) === propDF.col(s"${label.nodeLabel}_id"), "left_outer")
+            .drop(propDF.col(s"${label.nodeLabel}_id"))
+            .groupBy(tmpNodeDf.columns.map(tmpNodeDf.col): _*)
+            .agg(functions.collect_list(propDF.col(prop.propertyKey)).as(prop.propertyKey))
+
+          // this is necessary because Sparks CSV writer does not support array<string> data type
+          withListProperty.withColumn(prop.propertyKey, listToString(withListProperty.col(prop.propertyKey)))
+
+
```
Comment (Mats-SX) [inline spark-cypher/src/main/scala/org/opencypher/spark/tools/LdbcConverter.scala]: Some empty lines

File: spark-cypher/src/main/scala/org/opencypher/spark/tools/LdbcConverter.scala [hunk 3]
```diff
@@ -0,0 +1,577 @@
...
+        .map(_.get(0).toString)
+
+      newLabels.map(newLabel => Label(parseLabel(newLabel), Some(label)) -> nodeDf.where(typeColumn === lit(newLabel)).drop(typeColumn))
+
+    case (label, nodeDf) => Seq(label -> nodeDf)
+  }
+
+  private def splitRelDataFrames(rels: Map[RelType, DataFrame]): Map[RelType, DataFrame] = {
+
+
```
Comment (Mats-SX) [inline spark-cypher/src/main/scala/org/opencypher/spark/tools/LdbcConverter.scala]: empty line

File: spark-cypher/src/main/scala/org/opencypher/spark/tools/LdbcConverter.scala [hunk 4]
```diff
@@ -0,0 +1,577 @@
...
+    Label("Post") -> StructType(Seq(StructField("id", StringType, false), StructField("imageFile", StringType, true), StructField("creationDate", LongType, true), StructField("locationIP", StringType, true), StructField("browserUsed", StringType, true), StructField("language", StringType, true), StructField("content", StringType, true), StructField("length", StringType, true))),
+    Label("Comment") -> StructType(Seq(StructField("id", StringType, false), StructField("creationDate", LongType, true), StructField("locationIP", StringType, true), StructField("browserUsed", StringType, true), StructField("content", StringType, true), StructField("length", IntegerType, true))),
+    Label("TagClass") -> StructType(Seq(StructField("id", StringType, false), StructField("name", StringType, true), StructField("url", StringType, true))),
+    Label("Tag") -> StructType(Seq(StructField("id", StringType, false), StructField("name", StringType, true), StructField("url", StringType, true)))
+  )
+
+  val nodeDictionary: Map[Label, Int] = nodeMapping.zipWithIndex.map { case ((label, _), i) => label -> i }
```
Comment (Mats-SX) [inline spark-cypher/src/main/scala/org/opencypher/spark/tools/LdbcConverter.scala]: Could probably formulate this as `nodeMapping.keys.zipWithIndex.toMap`

File: spark-cypher/src/main/scala/org/opencypher/spark/tools/LdbcConverter.scala (hunk 1 above)
Comment (s1ck) [reply spark-cypher/src/main/scala/org/opencypher/spark/tools/LdbcConverter.scala]: This would result in `Map[(RelType, StructType), Int]`

Comment (Mats-SX) [reply spark-cypher/src/main/scala/org/opencypher/spark/tools/LdbcConverter.scala]: Are you sure? It worked on my machine™

Comment (s1ck) [reply spark-cypher/src/main/scala/org/opencypher/spark/tools/LdbcConverter.scala]: Oh of course. Didn't see the `.keys` call ... :1st_place_medal:

**Pull request closed without merging**